src/ConsentRequiredController.js

```diff
@@ -0,0 +1,135 @@
+/*!
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+define([
+  'okta',
+  'shared/util/Util',
+  'util/FormController',
+  'util/FormType',
+  'views/consent/ConsentHeader',
+  'views/consent/ConsentBeacon',
+  'views/consent/ScopeList',
+  'views/consent/ConsentButtonsBar'
+],
+function (Okta, Util, FormController, FormType, ConsentHeader, ConsentBeacon, ScopeList, ConsentButtonsBar) {
+
+  var $ = Okta.$,
+      _ = Okta._;
+
+  return FormController.extend({
+    className: 'consent-required',
+    initialize: function () {
+      this.model.set('expiresAt', this.options.appState.get('transaction').expiresAt);
```

Comment:
you shouldn't be accessing the transaction here. It is a better practice to create derived properties in AppState. Take a look at ```hasExistingPhones``` in there for an example